ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -22,6 +22,9 @@
 # _LIBRARIES, and _LINK_FLAGS.
 # If _INTERFACES is not empty it will be used exclusively, otherwise the other
 # variables are being used.
+# Packages exporting libraries in non-standard directories cannot rely on a
+# _LIBRARY_DIRS variable -- an absolute path to each library must be provided
```

Comment:
A CMake variable ending in `_LIBRARY_DIRS` is pretty uncommon I think. Did you maybe mean `_LIBRARY_DIR` instead?

How about checking if a `_LIBRARY_DIR` variable set and then join it with the library names (assuming these absolute paths exist)?

Reply:
Hmm, that's the variable that the original issue refers to. It's also the one that's populated by e.g. [`pkg-config` support macros](https://cmake.org/cmake/help/v3.5/module/FindPkgConfig.html). I cannot seem to find a use case online for a `_LIBRARY_DIR` variable, which one did you have in mind?

Reply:
No specific one. It is just the one I have seen in the past.

> How about checking if a _LIBRARY_DIR variable set and then join it with the library names (assuming these absolute paths exist)?

This would still be good to do. The logic would just need to consider all entries in `_LIBRARY_DIRS`.

Reply:
To search for the library you mean, right?

Reply:
Yes.